cdap-app-fabric/src/main/java/io/cdap/cdap/app/store/Store.java

```diff
@@ -292,10 +294,11 @@ Map<ProgramRunId, RunRecordDetail> getRuns(ApplicationId applicationId, ProgramR
    * Creates new application if it doesn't exist. Updates existing one otherwise.
    *
    * @param id            application id
-   * @param specification application specification to store
+   * @param meta          application metadata to store
+   * @param parentVersion version id of the application
    */
-  void addApplication(ApplicationId id, ApplicationSpecification specification);
-
+  void addApplication(ApplicationId id, ApplicationMeta meta, @Nullable String parentVersion)
```

Comment:
didn't put parentVersion in ApplicationMeta (provided explanation here https://github.com/cdapio/cdap/pull/14493/files#r984991613